test/e2e/pivoting.go

```diff
@@ -273,20 +287,21 @@ func installIronicBMO(inputGetter func() installIronicBMOInput) {
 	Expect(err).To(BeNil(), "Failed to deploy Ironic")
 }
 
-type RemoveIronicInput struct {
+type RemoveIronicBMOInput struct {
 	ManagementCluster framework.ClusterProxy
 	IsDeployment      bool
 	Namespace         string
 	NamePrefix        string
+	NameSuffix        string
 }
 
-func removeIronic(ctx context.Context, inputGetter func() RemoveIronicInput) {
+func removeIronicBMO(ctx context.Context, inputGetter func() RemoveIronicBMOInput) {
```

Comment:
combining ironic local and deployment deletion in one function was there  before. I am not adding that functionality here. I am adding BMO removal only. And since we know its always a deployment we always hardcode the `isDeployment` to true. Expecting common sense from developer here. Since a developer can also try to delete ironic deployment while it was deployed locally. One good point raised by you is the error message at line 304. We can add the deployment or component name to be deleted which I missed out .

Reply:
> combining ironic local and deployment deletion in one function was there before

I know, but adding BMO to this makes it harder to understand and easier to make mistakes. Splitting it into two functions was just a suggestion for how to make it cleaner, we can go the other way also. Then I would suggest to make it "smarter", similar to `installIronicBMO`. Add to the input a variable that tells if it is BMO or Ironic we are dealing with, then use it to set the other parameters that are already known, e.g. prefix/suffix. We also know that BMO is never deployed locally in this context so check and error out if we get this combination.

Reply:
I am in favor of keeping the function core as it is now as both ironic and bmo have the same lifecycle and do the refactoring later on?

Reply:
that might be this work will not solve the centos issue to delete bmh and then I would prefer to revert this changes

Reply:
I'm not understanding the issue. Why make a messy implementation that we have to fix in the future when we can make a nice implementation? Both of them can easily be reverted but only one have to be improved.

Just think about how the function is used. I want to remove BMO, ok `removeIronicBMO` sounds good. What does it take as input? Name suffix, what is this? Why should I have to know this to remove BMO? I just want to say if I want to remove BMO or Ironic.

I stand my ground. This must be simplified. Either split it into separate functions or make it "smarter" so it is harder to shoot yourself in the foot.

Reply:
> Just think about how the function is used. I want to remove BMO, ok `removeIronicBMO` sounds good. What does it take as input? Name suffix, what is this? Why should I have to know this to remove BMO? I just want to say if I want to remove BMO or Ironic.

I didn't get your point here, this is not asked from user. Theres bunch of other similar suffix, prefix parameter in the same file here. You as a developer dont need to know, you as a user will always give the name to delete no matter how simplified the function is. Suffix is a part of it.

Reply:
I can agree on one point , this might be a bit difficult to understand in future since ironic is deployed locally and in deployment while BMO is only deployed as a deployment. So for BMO isDeployment var doesn’t make any sense and leaves room for mistakes. Splitting it in two functions increases readability and understanding of the code.

Reply:
There is always a user. The user of the function is the developer. And no, the developer does *not* need to know the suffix and should not need to think about this. It is always `-controller-manager` for BMO and `-ironic` for Ironic. This can be taken care of by the function.

Alternatively, if you want to keep the flexibility of setting the exact name, then just make a generic function that can delete any deployment by giving a name. I'm also fine with that. But mixing it in one function is not nice.

Reply:
> There is always a user. The user of the function is the developer. And no, the developer does _not_ need to know the suffix and should not need to think about this. It is always `-controller-manager` for BMO and `-ironic` for Ironic. This can be taken care of by the function.


Hummm, I don’t think I agree. let’s agree to disagree here. Atleast we agree we need separate functions :)